pkg/services/ngalert/state/persister_sync.go

```diff
@@ -107,7 +107,7 @@ func (a *SyncStatePersister) saveAlertStates(ctx context.Context, states ...Stat
 			ResultFingerprint: s.ResultFingerprint.String(),
 		}
 
-		err = a.store.SaveAlertInstance(ctx, instance)
+		err = a.store.SaveAlertInstances(ctx, []models.AlertInstance{instance})
```

Comment:
It still saves instances one by one, and I didn’t want to change this because saving multiple instances at once in a long transaction might have negative consequences if there are many instances. Since it's just a refactoring, I'd rather keep the current behaviour.